src/zql/ast-to-ivm/pipeline-builder.test.ts

```diff
@@ -410,5 +410,132 @@ describe('OR', () => {
+    {op: 'IN', left: 'a', right: ['b', 'c'], expected: false},
+    {op: 'IN', left: true, right: [true, false], expected: true},
+    {op: 'IN', left: true, right: [false], expected: false},
+
+    {op: 'LIKE', left: 'abc', right: 'abc', expected: true},
+    {op: 'LIKE', left: 'abc', right: 'ABC', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'ab', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'ab%', expected: true},
+    {op: 'LIKE', left: 'abc', right: '%bc', expected: true},
+    {op: 'LIKE', left: 'abbc', right: 'a%c', expected: true},
+    {op: 'LIKE', left: 'abc', right: 'a_c', expected: true},
+    {op: 'LIKE', left: 'abc', right: 'a__', expected: true},
+    {op: 'LIKE', left: 'abc', right: '_bc', expected: true},
+    {op: 'LIKE', left: 'abc', right: '___', expected: true},
+    {op: 'LIKE', left: 'abc', right: '%', expected: true},
+    {op: 'LIKE', left: 'abc', right: '_', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'a', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'b', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'c', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'd', expected: false},
+    {op: 'LIKE', left: 'abc', right: 'ab', expected: false},
+
+    {op: 'ILIKE', left: 'abc', right: 'abc', expected: true},
+    {op: 'ILIKE', left: 'abc', right: 'ABC', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: 'ab', expected: false},
+    {op: 'ILIKE', left: 'Abc', right: 'ab%', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: '%bc', expected: true},
+    {op: 'ILIKE', left: 'Abbc', right: 'a%c', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: 'a_c', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: 'a__', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: '_bc', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: '___', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: '%', expected: true},
+    {op: 'ILIKE', left: 'Abc', right: '_', expected: false},
+    {op: 'ILIKE', left: 'Abc', right: 'a', expected: false},
+    {op: 'ILIKE', left: 'Abc', right: 'b', expected: false},
+    {op: 'ILIKE', left: 'Abc', right: 'c', expected: false},
+    {op: 'ILIKE', left: 'Abc', right: 'd', expected: false},
+    {op: 'ILIKE', left: 'Abc', right: 'ab', expected: false},
+
+    // and some tricky likes
+    {op: 'LIKE', left: 'abc', right: 'a%b%c', expected: true},
+    {op: 'LIKE', left: 'abc', right: 'a%b', expected: false},
+    {op: 'LIKE', left: 'abc', right: '.*', expected: false},
+    {op: 'LIKE', left: 'abc', right: '...', expected: false},
+    ...Array.from('/\\[](){}^$+?*.|%_', c => ({
+      op: 'LIKE',
+      left: c,
+      right: '_',
+      expected: true,
+    })),
+    {op: 'LIKE', left: 'a%b', right: 'a\\%b', expected: true},
+    {op: 'LIKE', left: 'a_b', right: 'a\\_b', expected: true},
+  ] as const;
+
+  for (const c of cases) {
+    test(`${c.left} ${c.op} ${c.right} === ${c.expected}`, () => {
```

Comment:
fwiw, you can use `test.each`
https://vitest.dev/api/#test-each

Reply:
I'll try to remember and change my pattern